File: frontend/types/post.ts
import { Comment } from "./comment";
import { User } from "./user";

export interface Post {
  id: number;
  user: User;
  user_id?: number;
  image?: string;
  caption?: string;
  privacy?: string;
  creation_date?: string;
  reactions?: Reaction;
  comments?: Comment[];
  comment_count?: number;
}

export interface Reaction {
  likes: number;
  dislikes: number;
  userReaction: boolean | null;
  isReacting?: boolean;
}

export interface PostShareModalProps {
  postId: number;
  isOpen: boolean;
  onClose: () => void;
}

File: backend/pkg/model/post.go
package model

type Post struct {
	ID           int            `json:"id"`
	UserId       int            `json:"user_id"`
	Author       string         `json:"author"`
	CategoryID   int            `json:"category_id"`
	Title        string         `json:"title"`
	Text         string         `json:"text"`
	CreationDate string         `json:"creation_date"`
	Comment      []*Comment     `json:"comments"`
	Privacy      string         `json:"privacy"`
	Caption      string         `json:"caption"`
	Image        string         `json:"image"`
	User         *User          `json:"user"`
	Reactions    ReactionCounts `json:"reactions"`
	CommentCount int            `json:"comment_count"`
}
